app/src/db/migrations/20210721210234_019-api-key-models.js

```diff
@@ -0,0 +1,170 @@
+const { v4: uuidv4 } = require('uuid');
+const stamps = require('../stamps');
+
+const CREATED_BY = 'migration-019';
+
+exports.up = function (knex) {
+  return Promise.resolve()
+    // Add the new permissions for users managing Form API Keys
+    .then(() => {
+      const items = [
+        {
+          createdBy: CREATED_BY,
+          code: 'form_api_create',
+          display: 'API Key Create',
+          description: 'Can create an api key for a form',
+          active: true
+        },
+        {
+          createdBy: CREATED_BY,
+          code: 'form_api_read',
+          display: 'API Key Read',
+          description: 'Can view the API key for a form (unencrypted)',
+          active: true
+        },
+        {
+          createdBy: CREATED_BY,
+          code: 'form_api_update',
+          display: 'API Key Update',
+          description: 'Can update the API key for a form',
+          active: true
+        },
+        {
+          createdBy: CREATED_BY,
+          code: 'form_api_delete',
+          display: 'API Key Delete',
+          description: 'Can remove the API key from a form',
+          active: true
+        }
+      ];
+      return knex('permission').insert(items).returning('code');
+    })
+
+    // Form Owners (and only owners now) can do these new permissions
+    .then(() => {
+      const rolePermssion = {
+        id: uuidv4(),
+        createdBy: CREATED_BY,
+        role: 'owner',
+        permission: 'form_api_create'
+      };
+      return knex('role_permission').insert(rolePermssion);
+    })
+
+    .then(() => {
+      const rolePermssion = {
+        id: uuidv4(),
+        createdBy: CREATED_BY,
+        role: 'owner',
+        permission: 'form_api_read'
+      };
+      return knex('role_permission').insert(rolePermssion);
+    })
+
+    .then(() => {
+      const rolePermssion = {
+        id: uuidv4(),
+        createdBy: CREATED_BY,
+        role: 'owner',
+        permission: 'form_api_update'
+      };
+      return knex('role_permission').insert(rolePermssion);
+    })
+
+    .then(() => {
+      const rolePermssion = {
+        id: uuidv4(),
+        createdBy: CREATED_BY,
+        role: 'owner',
+        permission: 'form_api_delete'
+      };
+      return knex('role_permission').insert(rolePermssion);
+    })
+
+    // Add the new table for API keys
+    .then(() => knex.schema.createTable('form_api_key', table => {
+      table.specificType(
+        'id',
+        'integer GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY'
+      );
+      table.uuid('formId').references('id').inTable('form').notNullable().index();
+      table.uuid('secret').unique().notNullable();
```

Comment:
Wondering if this column also needs to be indexed - there may be situations where we may need to reverse lookup from secret back to formId potentially?